File: maven-root/mtsa/src/main/java/MTSAEnactment/ar/uba/dc/lafhis/enactment/robot/N6/robotProtocol/TurnRightMessage.java
package MTSAEnactment.ar.uba.dc.lafhis.enactment.robot.N6.robotProtocol;

public class TurnRightMessage extends Message {	
	public static int MESSAGE_ID				= 206;
	
	public TurnRightMessage(int to, int from){
		super(MESSAGE_ID, 1, to, from);
	}
	
	public TurnRightMessage(String serializedMessage){
		super(serializedMessage);
	}
	
	@Override
	public int getMessageLength() {
		return super.getMessageLength();
	}
	
	
	@Override
	protected void readData(String inputString){
	}

	@Override
	protected String writeData(){
		return "";
	}

	@Override
	protected boolean hasReply() {
		return true;
	}	
}

File: maven-root/mtsa/src/main/java/RobotProtocol/src/robotProtocol/messages/TurnRightMessage.cpp
/*
 * TurnRightMessage.cpp
 *
 *  Created on: Apr 16, 2013
 *      Author: mariano
 */

#include "TurnRightMessage.h"

namespace RobotProtocol {

TurnRightMessage TurnRightMessage::staticMessage = TurnRightMessage(0,0);

Message *TurnRightMessage::deserialize(char *serializedMessage){
	TurnRightMessage::staticMessage.initialize(serializedMessage);
	return &(TurnRightMessage::staticMessage);
}

TurnRightMessage::TurnRightMessage(int to, int from)
:Message(MESSAGE_ID, 1, to, from){}

TurnRightMessage::TurnRightMessage(char *serializedMessage){
	initialize(serializedMessage);
}

TurnRightMessage::~TurnRightMessage() {
}

int TurnRightMessage::getMessageLength(){
	return Message::getMessageLength();
}

void TurnRightMessage::readData(char **serializedMessageIndex){}

void TurnRightMessage::writeData(char **serializedMessageIndex){}



} /* namespace RobotProtocol */
